services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jFlowSegmentRepository.java

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories.impl;
+
+import org.openkilda.model.FlowSegment;
+import org.openkilda.persistence.repositories.FlowSegmentRepository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Neo4J OGM implementation of {@link FlowSegmentRepository}.
+ */
+public class Neo4jFlowSegmentRepository extends Neo4jGenericRepository<FlowSegment> implements FlowSegmentRepository {
+    public Neo4jFlowSegmentRepository(Neo4jSessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public Iterable<FlowSegment> findByFlowId(String flowId) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("flow_id", flowId);
+
+        return getSession().query(FlowSegment.class,
+                "MATCH (src:switch)-[fs:flow_segment{flowid: {flow_id}}]->(dst:switch) RETURN fs, src, dst",
```

Comment:
Is there any reason to return `src` and `dst`?

Reply:
Yes, OGM requires them as FlowSegment has srcSwitch and destSwitch properties.